plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/ui/views/log/LogView.java

```diff
@@ -165,27 +119,57 @@ public void restoreLog() {
 		activate();
 		getViewTree();
 		Menu cm = new ContextMenu(RESTORE_LOG);
-		cm.select();			
+		cm.select();
 	}
-	
 
-	private void setFilter(String severity){
-		ViewMenu tmenu = new ViewMenu("Filters...");
-		tmenu.select();
-		new WaitUntil(new ShellWithTextIsAvailable("Log Filters"));
-		new DefaultShell("Log Filters");
-		new CheckBox(OK_SEVERITY).toggle(false);
-		new CheckBox(INFORMATION_SEVERITY).toggle(false);
-		new CheckBox( WARNING_SEVERITY).toggle(false);
-		new CheckBox(ERROR_SEVERITY).toggle(false);
-		new CheckBox(severity).toggle(true);
-		new CheckBox("Limit visible events to:").toggle(false);
-		new PushButton("OK").click();
-		new GroupWait(waitWhile(new ShellWithTextIsAvailable("Log Filters")),
-				waitWhile(new ShellWithTextIsAvailable("Progress Information")));
-	}
-	
 	private Tree getViewTree(){
 		return new DefaultTree(cTabItem);
 	}
+
+	private List<LogMessage> getMessages(Severity severity){
+		activate();
+
+		ArrayList<LogMessage> messages = new ArrayList<>();
+		for(TreeItem item : new DefaultTree().getItems()){
```

Comment:
LogViewTest.testClearAndRestore failed. Maybe there is a possibility that Tree does not exist when LogView has no messages ?